mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java

```diff
@@ -89,7 +95,7 @@ public void event(Event event) {
                     .stream()
                     .map(Throwing.function(MailboxMessage::getFullContent))
                     .collect(Guavate.toImmutableList());
-                spamAssassin.learnHam(contents, addedEvent.getSession().getUser().getUserName());
+                spamAssassin.learnHam(contents, addedEvent.getUser().getLocalPart());
```

Comment:
I replaced all `getlocalPart()` by `asString()`. Sorry I squashed changes into commit history to avoid long long history of fixup. the length of commits is already long.